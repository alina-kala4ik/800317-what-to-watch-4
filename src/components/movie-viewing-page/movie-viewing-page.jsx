import React, {Component, createRef} from "react";
import PropTypes from "prop-types";
import history from "./../../history.js";

class MovieViewingPage extends Component {
  constructor(props) {
    super(props);

    this.playerRef = createRef();

    this.handleFullScreenClick = this.handleFullScreenClick.bind(this);
  }

  componentDidUpdate(prevProps) {
    const {isFullScreenMode} = this.props;
    const player = this.playerRef.current;

    if (this.props.isFullScreenMode !== prevProps.isFullScreenMode) {
      if (isFullScreenMode) {
        player.requestFullscreen();
      } else {
        document.exitFullscreen();
      }
    }
  }

  handleFullScreenClick() {
    const {onFullScreenClick} = this.props;
    onFullScreenClick();
  }

  handleExitClick() {
    history.goBack();
  }

  render() {
    const {isPlaying, progress, timeLeft, children, onPlayClick, onPauseClick} = this.props;

    return <div
      className="player"
      ref={this.playerRef}
    >
      {children}

      <button
        type="button"
        className="player__exit"
        onClick={this.handleExitClick}
      >
        Exit
      </button>

      <div
        className="player__controls"
      >
        <div className="player__controls-row">
          <div className="player__time">
            <progress className="player__progress" value={progress} max="100"></progress>
            <div className="player__toggler" style={{left: `${progress}%`}}>Toggler</div>
          </div>
          <div className="player__time-value">{timeLeft}</div>
        </div>

        <div className="player__controls-row">
          {isPlaying === false &&
            <button
              type="button"
              className="player__play"
              onClick={onPlayClick}
            >
              <svg viewBox="0 0 19 19" width="19" height="19">
                <use xlinkHref="#play-s"></use>
              </svg>
              <span>Play</span>
            </button>
          }

          {isPlaying === true &&
            <button
              type="button"
              className="player__play"
              onClick={onPauseClick}
            >
              <svg viewBox="0 0 14 21" width="14" height="21">
                <use xlinkHref="#pause"></use>
              </svg>
              <span>Pause</span>
            </button>
          }

          <div className="player__name">Transpotting</div>

          <button
            type="button"
            className="player__full-screen"
            onClick={this.handleFullScreenClick}
          >
            <svg viewBox="0 0 27 27" width="27" height="27">
              <use xlinkHref="#full-screen"></use>
            </svg>
            <span>Full screen</span>
          </button>
        </div>
      </div>

    </div>;
  }
}

MovieViewingPage.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  progress: PropTypes.number.isRequired,
  timeLeft: PropTypes.string,
  children: PropTypes.node.isRequired,
  onPlayClick: PropTypes.func.isRequired,
  onPauseClick: PropTypes.func.isRequired,
  onFullScreenClick: PropTypes.func.isRequired,
  isFullScreenMode: PropTypes.bool.isRequired,
};

export default MovieViewingPage;

