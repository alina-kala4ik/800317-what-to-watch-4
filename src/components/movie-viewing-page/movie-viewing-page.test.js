import React from "react";
import renderer from "react-test-renderer";
import {MovieViewingPage} from "./movie-viewing-page.jsx";

describe(`render MovieViewingPage`, ()=>{
  it(`render MovieViewingPage with player controls and during play`, ()=>{
    const tree = renderer.create(
        <MovieViewingPage
          onExitClick={()=>{}}
          isPlaying={true}
          progress={0}
          timeLeft={`01:00:00`}
          isControllersVisible={true}
          onPlayClick={()=>{}}
          onPauseClick={()=>{}}
          onFullScreenClick={()=>{}}
          onMouseEnterControls={()=>{}}
          onMouseLeaveControls={()=>{}}
        >
          <video />
        </MovieViewingPage>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render MovieViewingPage with player controls and during pause`, ()=>{
    const tree = renderer.create(
        <MovieViewingPage
          onExitClick={()=>{}}
          isPlaying={false}
          progress={0}
          timeLeft={`01:00:00`}
          isControllersVisible={true}
          onPlayClick={()=>{}}
          onPauseClick={()=>{}}
          onFullScreenClick={()=>{}}
          onMouseEnterControls={()=>{}}
          onMouseLeaveControls={()=>{}}
        >
          <video />
        </MovieViewingPage>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render MovieViewingPage without player controls`, ()=>{
    const tree = renderer.create(
        <MovieViewingPage
          onExitClick={()=>{}}
          isPlaying={true}
          progress={0}
          timeLeft={`01:00:00`}
          isControllersVisible={false}
          onPlayClick={()=>{}}
          onPauseClick={()=>{}}
          onFullScreenClick={()=>{}}
          onMouseEnterControls={()=>{}}
          onMouseLeaveControls={()=>{}}
        >
          <video />
        </MovieViewingPage>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
