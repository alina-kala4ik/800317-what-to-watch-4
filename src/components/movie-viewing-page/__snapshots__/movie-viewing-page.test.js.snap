// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieViewingPage render MovieViewingPage with player controls and during pause 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "opacity": "1",
      }
    }
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "opacity": "1",
      }
    }
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        01:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`render MovieViewingPage render MovieViewingPage with player controls and during play 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "opacity": "1",
      }
    }
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "opacity": "1",
      }
    }
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        01:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`render MovieViewingPage render MovieViewingPage without player controls 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "opacity": "0",
      }
    }
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "opacity": "0",
      }
    }
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        01:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
